### PR TITLE
rowdy router refactor

### PR DESCRIPTION
**Pull request by paul-jacob**
smartcitiesdata/smartcitiesdata/discovery_ui#131

co-authored-by: Paul Linville <plinville@pillartechnology.com>

### CODE REVIEW

File: config/config-emacs-elixir-format.exs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+use Mix.Config
```
Comment (ManApart) [inline config/config-emacs-elixir-format.exs]: Why do we have an emacs specific elixir format config?

File: lib/discovery_api_web/plugs/restrictor.ex [hunk 2]
```diff
@@ -2,14 +2,14 @@ defmodule DiscoveryApiWeb.Plugs.Restrictor do
   @moduledoc false
   require Logger
   import Plug.Conn
-  alias DiscoveryApiWeb.Utilities.AuthUtils
+  alias DiscoveryApiWeb.Services.AuthService
```
Comment (ManApart) [inline lib/discovery_api_web/plugs/restrictor.ex]: I appreciate this rename.

File: lib/discovery_api_web/services/dataset_query_service.ex [hunk 3]
```diff
@@ -1,4 +1,4 @@
-defmodule DiscoveryApi.DatasetQueryService do
+defmodule DiscoveryApiWeb.DatasetQueryService do
```
Comment (ManApart) [inline lib/discovery_api_web/services/dataset_query_service.ex]: I have no dog in this fight. Why `DiscoveryApiWeb` or `DiscoveryApi`?

File: lib/discovery_api_web/router.ex [hunk 4]
```diff
@@ -4,91 +4,49 @@ defmodule DiscoveryApiWeb.Router do
   """
```
Comment (ManApart) [inline lib/discovery_api_web/router.ex]: This looks a lot better!

File: config/config-emacs-elixir-format.exs (hunk 1 above)
Comment (jdenen) [reply config/config-emacs-elixir-format.exs]: As much as I love Emacs, I don't think this is right. 🤔

File: lib/discovery_api_web/services/auth_service.ex [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+defmodule DiscoveryApiWeb.Services.AuthService do
   @moduledoc """
-  A service wrapper for paddle to do dn lookups and extract user id
+  Provides authentication and authorization helper methods
   """
 
-  def get_members(org_dn) do
+  alias DiscoveryApi.Data.Model
+
+  def has_access?(%Model{private: false} = _dataset, _username), do: true
```
Comment (jdenen) [inline lib/discovery_api_web/services/auth_service.ex]:
```elixir
  @type dataset() :: %Model{}
  @type username() :: String.t() | nil

  @spec has_access?(dataset(), username())
  def has_access?(%Model{private: false}, _), do: true
```

I prefer using specs to express variable names that are often pattern matched out. I think it reads better/reduces noise. Just a thought, though. I know everyone won't agree.

File: lib/discovery_api_web/services/auth_service.ex [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
+
+  def has_access?(%Model{private: true} = _dataset, nil), do: false
```
Comment (jdenen) [inline lib/discovery_api_web/services/auth_service.ex]:
```elixir
  def has_access?(%Model{private: true}, nil), do: false
```

If you did the spec thing above.

File: lib/discovery_api_web/services/auth_service.ex [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
+
+  def has_access?(%Model{private: true, organizationDetails: %{dn: dn}} = _dataset, username) do
```
Comment (jdenen) [inline lib/discovery_api_web/services/auth_service.ex]:
```elixir
  def has_access?(%Model{private: true, organizationDetails: %{dn: dn}}, username) do
```

Same here.

File: lib/discovery_api_web/services/auth_service.ex [hunk 8]
```diff
@@ -1,9 +1,23 @@
...
+    dn
+    |> get_members()
+    |> Enum.member?(username)
+  end
+
+  def has_access?(_base, _case), do: false
```
Comment (jdenen) [inline lib/discovery_api_web/services/auth_service.ex]:
```elixir
  def has_access?(_, _), do: false
```

Same here.

File: lib/discovery_api_web/utilities/s3_wrapper.ex [hunk 9]
```diff
@@ -0,0 +1,65 @@
+defmodule DiscoveryApiWeb.Utilities.S3Wrapper do
```
Comment (jdenen) [inline lib/discovery_api_web/utilities/s3_wrapper.ex]: You probably just moved this file, but you're getting this feedback anyway.

Can we rename this module? I know for the moment we're setting up against S3 specifically, but we want this to be agnostic. Something like `BucketWrapper` or `ObjectStorageWrapper`?

File: lib/discovery_api_web/plugs/restrictor.ex (hunk 2 above)
Comment (paul-jacob) [reply lib/discovery_api_web/plugs/restrictor.ex]: This is actually wrong

File: lib/discovery_api_web/utilities/s3_wrapper.ex (hunk 9 above)
Comment (bbrewer-papa-x) [reply lib/discovery_api_web/utilities/s3_wrapper.ex]: Will do

File: lib/discovery_api_web/services/auth_service.ex (hunk 5 above)
Comment (ManApart) [reply lib/discovery_api_web/services/auth_service.ex]: @jdenen I haven't formed an opinion on this. I'd like to talk with you about it; maybe talk about it as arts and crafts after that.

**Pull request merged**